maven/bnd-maven-plugin/src/main/java/aQute/bnd/maven/plugin/AbstractBndMavenPlugin.java

```diff
@@ -154,6 +161,14 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 			return;
 		}
 
+		File classesDir = getClassesDir();
+
+		if (includeClassesDir && isEmpty(classesDir) && skipIfEmpty) {
```

Comment:
I would test `skipIfEmpty` first to possibly avoid the work of `isEmpty(classesDir)`.